### PR TITLE
Fix: closes  #505. 'ordering' alias my clash with a field named 'orde…

### LINKED ISSUE #505: ordering alias clash
AutocompleteModel.order_choices use an extra construct that injects (CASE ... END) AS ordering, that may clash with your model if you happen to have a field named 'ordering'.

I'll fix it in my fork and ask for  a pull in a while. I'd used '_ordering' instead of 'ordering' and make it  orvewritable with an attribute names ordering_alias.

sandro
*:-)

### PR DESCRIPTION
**Pull request by sandroden**
please let me know if you prefere to use different names of you prefere to place docs somewhere else. 

sandro
*:-)

### CODE REVIEW

File: autocomplete_light/autocomplete/model.py [hunk 1]
```diff
@@ -44,6 +43,15 @@ class AutocompleteModel(object):
         single field name or an iterable (ie. list, tuple).
         However, if AutocompleteModel is instanciated with a list of values,
         it'll reproduce the ordering of values.
+
+    .. py:attribute:: ordering_alias
+
+        Internal attribute that determines which alias is used by
+        injected sql that will show selected items first. Default is
+        _ordering that is probably very unusual as field_name. You
+        shuld worry to set this attribute only if you have an
```
Comment (jpic) [inline autocomplete_light/autocomplete/model.py]: shuld/should

File: autocomplete_light/autocomplete/model.py [hunk 2]
```diff
@@ -83,14 +91,15 @@ def order_choices(self, choices):
                 for i, pk in enumerate(self.values)])
             ordering = 'CASE %s END' % clauses
 
-            _order_by = ('ordering',)
+            ordering_alias = getattr(self, 'ordering_alias', '_ordering')
```
Comment (jpic) [inline autocomplete_light/autocomplete/model.py]: Why has the default changed from `ordering` to `_ordering` here ?

**Comment by jpic** [comment]
Thanks @sandroden !

We need a unit test for this, to garantee there won't be any regression in your feature ;)

Thanks !

**Comment by sandroden** [comment]
On Wed, Oct 07, 2015 at 09:01:09AM -0700, James Pic wrote:
>    Thanks [1]@sandroden !
>
>    We need a unit test for this, to garantee there won't be any regression
>    in your feature ;)

ok, sorry. I thought I would do that this week but I was really too busy. I
hope next week will be better...

*;-)

**Comment by jpic** [comment]
Same here xD

**Comment by jpic** [comment]
We're going to do another release in the next 2 to 3 weeks, could you please add some regression tests ? I really don't want to make another release without your changes in !!

Thanks a heap !!

**Comment by sandroden** [comment]
On Wed, Nov 18, 2015 at 04:33:35PM -0800, James Pic wrote:
>    We're going to do another release in the next 2 to 3 weeks, could you
>    please add some regression tests ? I really don't want to make another
>    release without your changes in !!

ok, i'll really try to do it  next week. Thanks for pinging!...

sandro
*:-)

File: autocomplete_light/autocomplete/model.py [hunk 3]
```diff
@@ -44,6 +43,15 @@ class AutocompleteModel(object):
...
         However, if AutocompleteModel is instanciated with a list of values,
         it'll reproduce the ordering of values.
+
+    .. py:attribute:: ordering_alias
+
+        Internal attribute that determines which alias is used by
+        injected sql that will show selected items first. Default is
+        _ordering that is probably very unusual as field_name. You
+        shuld worry to set this attribute only if you have an
+        attribute on db table named '_ordering'
```
Comment (jpic) [inline autocomplete_light/autocomplete/model.py]: Does it really make sense to change the "ordering" alias here ? Shouldn't we just use a less-conflicting name like `_autocomplete_light_ordering_` or something like that ?

Comment (sandroden) [inline autocomplete_light/autocomplete/model.py]: On Sun, Dec 20, 2015 at 03:21:52PM -0800, James Pic wrote:
>    In [1]autocomplete_light/autocomplete/model.py:
> > @@ -44,6 +43,15 @@ class AutocompleteModel(object):
> >          single field name or an iterable (ie. list, tuple).
> >          However, if AutocompleteModel is instanciated with a list of values,
> >          it'll reproduce the ordering of values.
> > +
> > +    .. py:attribute:: ordering_alias
> > +
> > +        Internal attribute that determines which alias is used by
> > +        injected sql that will show selected items first. Default is
> > +        _ordering that is probably very unusual as field_name. You
> > +        shuld worry to set this attribute only if you have an
> > +        attribute on db table named '_ordering'
>
>    Does it really make sense to change the "ordering" alias here ?
>    Shouldn't we just use a less-conflicting name like
>    _autocomplete_light_ordering_ or something like that ?


I agree. Really I don't think the '_ordering' name is normally used,
but with your proposal we would be completely sure!

I can't modify it before late tomorrow, thougt.

sandro
*:-)

Comment (sandroden) [inline autocomplete_light/autocomplete/model.py]: On Sun, Dec 20, 2015 at 03:21:52PM -0800, James Pic wrote:
>    In [1]autocomplete_light/autocomplete/model.py:
> > @@ -44,6 +43,15 @@ class AutocompleteModel(object):
> >          single field name or an iterable (ie. list, tuple).
> >          However, if AutocompleteModel is instanciated with a list of values,
> >          it'll reproduce the ordering of values.
> > +
> > +    .. py:attribute:: ordering_alias
> > +
> > +        Internal attribute that determines which alias is used by
> > +        injected sql that will show selected items first. Default is
> > +        _ordering that is probably very unusual as field_name. You
> > +        shuld worry to set this attribute only if you have an
> > +        attribute on db table named '_ordering'
>
>    Does it really make sense to change the "ordering" alias here ?
>    Shouldn't we just use a less-conflicting name like
>    _autocomplete_light_ordering_ or something like that ?


Done! Tests pass. Ready to merge


sandro
*:-)

File: autocomplete_light/fixtures/autocomplete_model_test_case.json [hunk 4]
```diff
@@ -67,7 +71,8 @@
     "fields": {
         "username": "Elton",
         "groups": [],
-        "email": "elton@example.com"
+        "email": "elton@example.com",
+	"ordering": 10
```
Comment (jpic) [inline autocomplete_light/fixtures/autocomplete_model_test_case.json]: I don't want to be "that guy" and complain about indentation here, it's, after all, JSON and it doesn't matter ^^

**Comment by jpic** [comment]
Thanks, I really want this to be in the 2.3.0 release in January because I don't want you to have to re-apply your patch in the next releases and also that's a pretty major bugfix I'd say ^^ But do you think you could rebase your comits to get rid of the merge commit ? `git rebase -i origin/master` are you familiar with that ?

**Comment by sandroden** [comment]
On Wed, Dec 23, 2015 at 03:33:43PM -0800, James Pic wrote:
>    Thanks, I really want this to be in the 2.3.0 release in January
>    because I don't want you to have to re-apply your patch in the next
>    releases and also that's a pretty major bugfix I'd say ^^ But do you
>    think you could rebase your comits to get rid of the merge commit
>    ?

ok

>    git rebase -i origin/master are you familiar with that ?

not really... I guess you want that my changes to be rebased on your
last commit ( upstream for me), i.e.: after the merge. Should that be

  git rebase - origin/master

do the trick. It didn't for me. The editor opened with "noop" instead
of the list of the "changeset" I would imagine...

Any help is very appreciated

sandro
*:-)

**Comment by jpic** [comment]
Did you try with `-i` ? as in `git rebase -i origin/master`

**Comment by blueyed** [comment]
> Did you try with -i ?

Probably, since the editor was opened.

@sandroden 
It should be `git rebase -i upstream/master` if you're using `upstream` for the main repo.

**Comment by sandroden** [comment]
On Fri, Dec 25, 2015 at 05:51:52AM -0800, Daniel Hahler wrote:
>      Did you try with -i ?
>
>    Probably, since the editor was opened.

Correct. My stesp pretty reprocuceable:

  git remote upstream https://github.com/yourlabs/django-autocomplete-light.git
  git remote add upstream https://github.com/yourlabs/django-autocomplete-light.git
  git rebase -i origin/master

That opens emacs w/ just "noop"  where the list of changesets should
be. I don't know waht it is... do you have a different behaviour for
the reported commands?


sandro
*:-)

**Comment by jpic** [comment]
I find this short video a good introduction:
https://www.youtube.com/watch?v=6EiI5XgtvFU
​
Thanks to @blueyed for teaching this technique to me last year ;)

**Comment by blueyed** [comment]
@sandroden 
See my previous comment: you want to rebase on `upstream/master`, not `origin/master` (which is your fork).

You are only seeing "noop", because you _are_ on "origin/master" already, so there's nothing to rebase.

(FWIW, I find it better to have my fork/remote named after my nick, instead of using `origin` for it.)

**Comment by jpic** [comment]
Closing in favor of #506 (rebased)

**Pull request closed without merging**